### PR TITLE
Upgrade aws-node to v1.4.1

### LINKED ISSUE #734: Use latest aws-node add-on

### PR DESCRIPTION
**Pull request by errordeveloper**
### Description

Closes #734.

### Checklist

- [x] Code compiles correctly (i.e `make build`)
- [x] Added tests that cover your change (if possible)
- [x] All unit tests passing (i.e. `make test`)

### CODE REVIEW

**Comment by errordeveloper** [comment]
@tiffanyfay @christopherhein FYI.

**Comment by errordeveloper** [comment]
Looks like image is not yet available in `eu-west-2`.

**Comment by christopherhein** [comment]
Awesome! we’re not making 1.4 default in EKS for another 1-2 weeks so good to get this primed.

**Comment by errordeveloper** [comment]
> we’re not making 1.4 default in EKS for another 1-2 weeks so good to get this primed

Is this information available in a place where we can track it?

**Comment by StevenACoffman** [comment]
aws/containers-roadmap#149

**Comment by errordeveloper** [comment]
@christopherhein @tiffanyfay any updates on 1.4 rollout?

**Comment by errordeveloper** [comment]
I've tested and it works now.

**Comment by StevenACoffman** [comment]
[1.4.1 just released](https://github.com/aws/amazon-vpc-cni-k8s/blob/master/config/v1.4/aws-k8s-cni.yaml#L73), btw.

File: pkg/addons/default/assets/aws-node.yaml [hunk 1]
```diff
@@ -64,12 +64,13 @@ spec:
...
         scheduler.alpha.kubernetes.io/critical-pod: ''
     spec:
+      priorityClassName: system-node-critical
       serviceAccountName: aws-node
       hostNetwork: true
       tolerations:
       - operator: Exists
       containers:
-      - image: 602401143452.dkr.ecr.us-west-2.amazonaws.com/amazon-k8s-cni:v1.3.2
+      - image: 602401143452.dkr.ecr.us-west-2.amazonaws.com/amazon-k8s-cni:v1.4.0
```
Comment (StevenACoffman) [inline pkg/addons/default/assets/aws-node.yaml]:
```suggestion
      - image: 602401143452.dkr.ecr.us-west-2.amazonaws.com/amazon-k8s-cni:v1.4.1
```

File: pkg/addons/default/assets/aws-node-1.10.yaml [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+        k8s-app: aws-node
+      annotations:
+        scheduler.alpha.kubernetes.io/critical-pod: ''
+    spec:
+      serviceAccountName: aws-node
+      hostNetwork: true
+      tolerations:
+      - operator: Exists
+      containers:
+      - image: 602401143452.dkr.ecr.us-west-2.amazonaws.com/amazon-k8s-cni:v1.4.0
```
Comment (StevenACoffman) [inline pkg/addons/default/assets/aws-node-1.10.yaml]:
```suggestion
      - image: 602401143452.dkr.ecr.us-west-2.amazonaws.com/amazon-k8s-cni:v1.4.1
```

File: pkg/addons/default/assets/aws-node.yaml (hunk 1 above)
Comment (errordeveloper) [reply pkg/addons/default/assets/aws-node.yaml]: This one is actually generated, so I need to update the URLs here:

https://github.com/weaveworks/eksctl/blob/312ba0cef113a6f98d1e0743eb0df161aebcdf98/pkg/addons/default/generate.go#L3-L4

And run `make generate`.

**Pull request merged**